slc/build.sh

```diff
@@ -157,11 +224,57 @@ if [ ! -f "$STUDIO_ADAPTER_PACK_PATH/apack.json" ]; then
 fi
 
 if [ "$skip_gen" = false ]; then
-    slc generate -d $OUTPUT_DIR $PROJECT_FLAG $SILABS_APP_PATH --with $SILABS_BOARD $CONFIG_ARGS --generator-timeout=3500 -o makefile
-    if [ $? -ne 0 ]; then
-        echo "FAILED TO Generate : $SILABS_APP_PATH"
-        exit 1
+    if [[ "$SILABS_APP_PATH" == *.slcw ]]; then 
+        # WiFi SOC solution       
+        if [[ "$SILABS_APP_PATH" == *917-soc* ]]; then
+            # Get application args
+            APP_WITH_ARG=$(build_with_arg "$SILABS_BOARD" "$WITH_BOOTLOADER_COMPONENTS")
+            APP_WITHOUT_ARG=$(build_without_arg "$WITHOUT_BOOTLOADER_COMPONENTS")
+            
+            echo "Generating application..."
+            slc generate --tt -s $GSDK_ROOT --daemon -d $OUTPUT_DIR $PROJECT_FLAG $SILABS_APP_PATH $APP_WITH_ARG $APP_WITHOUT_ARG $CONFIG_ARGS --generator-timeout=1800
+            if [ $? -ne 0 ]; then
+                echo "FAILED TO Generate application for: $SILABS_APP_PATH"
+                exit 1
+            fi
+        # Other solutions
+        else            
+            # Get bootloader arguments
+            BOOTLOADER_WITH_ARG=$(build_with_arg "$SILABS_BOARD" "$WITH_BOOTLOADER_COMPONENTS")
+            BOOTLOADER_WITHOUT_ARG=$(build_without_arg "$WITHOUT_BOOTLOADER_COMPONENTS")
+            
+            # Get application arguments
+            APPLICATION_WITH_ARG=$(build_with_arg "$SILABS_BOARD" "$WITH_APP_COMPONENTS")
+            APPLICATION_WITHOUT_ARG=$(build_without_arg "$WITHOUT_APP_COMPONENTS")
+           
+            # Generate bootloader first
+            echo "Generating bootloader..."    
+            slc generate --tt -s $GSDK_ROOT --daemon -d $OUTPUT_DIR $PROJECT_FLAG $SILABS_APP_PATH $BOOTLOADER_WITH_ARG $BOOTLOADER_WITHOUT_ARG -pids bootloader $CONFIG_ARGS --generator-timeout=1800
+            if [ $? -ne 0 ]; then
+                echo "FAILED TO Generate bootloader for: $SILABS_APP_PATH"
+                exit 1
+            fi
+            
+            # Generate application 
+            echo "Generating application..."
+            slc generate --tt -s $GSDK_ROOT --daemon -d $OUTPUT_DIR $PROJECT_FLAG $SILABS_APP_PATH $APPLICATION_WITH_ARG $APPLICATION_WITHOUT_ARG -pids application $CONFIG_ARGS --generator-timeout=1800
+            if [ $? -ne 0 ]; then
+                echo "FAILED TO Generate application for: $SILABS_APP_PATH"
+                exit 1
+            fi
+        fi
+    else
+        # Generate .slcp projects
+        WITH_ARG=$(build_with_arg "$SILABS_BOARD" "$WITH_APP_COMPONENTS" "Application")
+        WITHOUT_ARG=$(build_without_arg "$WITHOUT_APP_COMPONENTS")
```

Comment:
What happens if the slcp is a bootloader? Consummers of the script could try to add components with those arguments which would not be used in the generation.
To avoid this case and simplify the logic, we can limit the `*_app` and `*_bootloader` to only be used with solutions.

I'm open to other solution but the current implementation will silently fail.